File: src/main/kotlin/dev/quozul/database/helpers/ApiServer.kt
package dev.quozul.database.helpers

import dev.quozul.database.enums.ServerType
import kotlinx.serialization.Serializable

@Serializable
data class ApiServer(
	val version: String,
	val serverType: ServerType,
	val useAikar: Boolean,
	val jvmFlags: String?,
	val forgeVersion: String?,
	val fabricLauncherVersion: String?,
	val fabricLoaderVersion: String?,
	val quiltLauncherVersion: String?,
	val quiltLoaderVersion: String?,
	val ftbModpackId: Int?,
	val ftbModpackVersionId: Int?,
)

File: front/src/components/api/models/ApiServer.ts
export type ApiServer = {
	version: string;
	eula: boolean;
	serverType: string;
	useAikar: boolean;
	jvmFlags: string | null;
	memory: string;
	forgeVersion: string | null;
	fabricLauncherVersion: string | null;
	fabricLoaderVersion: string | null;
	quiltLauncherVersion: string | null;
	quiltLoaderVersion: string | null;
	ftbModpackId: number | null;
	ftbModpackVersionId: number | null;
};
